Gems/Atom/RHI/Code/Tests/Factory.cpp

```diff
@@ -163,12 +163,12 @@ namespace UnitTest
 
     AZ::RHI::Ptr<AZ::RHI::IndirectBufferSignature> Factory::CreateIndirectBufferSignature()
     {
-        return aznew IndirectBufferSignature;
+        return aznew NiceIndirectBufferSignature;
```

Comment:
NiceIndirectBufferSignature? Is this a typo?

Reply:
No, these are `using` declarations which mockup the test-version of `IndirectBufferSignature`, which is used as
```cpp
using NiceIndirectBufferSignature = ::testing::NiceMock<IndirectBufferSignature>;
```

Since we need the factory and the test code at the moment for the `IndirectBufferSignatureTests` just manually allocates `NiceIndirectBufferSignature`, this fix is needed so that it also properly works then with the multi-device version, which calls the factory methods (and then would not return the correct type).

Reply:
Ahh. Thanks for explaining.